src/chunked_scatter/chunked_scatter.py

```diff
@@ -81,13 +83,19 @@ def chunked_scatter(regions: Iterable[BedRegion],
         if contigs_can_be_split or chunk.contig != current_contig:
             current_contig = chunk.contig
             # and the current bed file contains enough bases
-            if current_scatter_size >= minimum_base_pairs:
+            minimum_reached = current_scatter_size >= list_size
+            maximum_reached = current_scatter_size + len(chunk) >= list_size
+            # Yield if the minimum is reached, or if current chunk will
+            # overflow the size and there is at least one chunk already.
+            if (minimum_reached and not size_is_maximum) or (
+                    maximum_reached and size_is_maximum and len(chunk_list) > 0
+            ):
```

Comment:
The if feel like it contains a bit too much (I'm not a big fan of having a bunch of `and`s and `or`s in the one expression).

```suggestion
            # Yield if the minimum is reached, or if current chunk will
            # overflow the size and there is at least one chunk already.
            size_to_check = current_scatter_size + len(chunk) if size_is_maximum else current_scatter_size
            if size_to_check >= list_size and len(chunk_list) > 0:
```

Reply:
Much better indeed :+1: